src/IECoreScene/MeshAlgoTriangulate.cpp

```diff
@@ -324,34 +259,17 @@ struct TriangulateFn
 	};
 };
 
-void reportError( const std::string &context, const std::string &message, bool throwExceptions )
-{
-	if ( throwExceptions )
-	{
-		throw InvalidArgumentException( boost::str ( boost::format("%s : %s") % context % message ) );
-	}
-	else
-	{
-		msg( MessageHandler::Level::Error, context, message );
-	}
-}
-
 } // namespace
 
 MeshPrimitivePtr MeshAlgo::triangulate(
-	const MeshPrimitive *mesh, float tolerance, bool throwExceptions
+	const MeshPrimitive *mesh
 )
 {
 	MeshPrimitivePtr meshCopy = mesh->copy();
 
 	if ( !mesh->arePrimitiveVariablesValid() )
 	{
-		reportError( "MeshAlgo::triangulate", "Mesh with invalid primitive variables ", throwExceptions );
-
-		if ( !throwExceptions )
-		{
-			return nullptr;
-		}
+		throw InvalidArgumentException( "MeshAlgo::triangulate : Mesh with invalid primitive variables" );
```

Comment:
Looking at where this called, I don't think there is any change in behaviour actually.  You're talking about where `triangulate` is called from `ToGLMeshConverter::doConversion`, right?  `ToGLMeshConverter::doConversion` starts by checking if P exists, and throwing an exception already before it gets into `triangulate`.  It may be true that Gaffer needs to handle this better, but this code change won't make any difference ( I guess technically it does make a difference in the case of `!arePrimitiveVariablesValid()`, but that is less likely than P being deleted ).

Reply:
> I guess technically it does make a difference in the case of !arePrimitiveVariablesValid()

Yep, that's why I hung my comment on the check for invalid primitive variables :) I have actually seen this warning in the wild fairly recently, so I don't think it's purely hypothetical. If you could look into improving the handling in Gaffer that would be great.